### DISCUSSION TITLE
AcroForm field Label

### DISCUSSION THREAD

**Discussion by Waqarcode** [py-pdf/pypdf#1024] [Q&A]
![image](https://user-images.githubusercontent.com/93347841/175525812-92d8b4b4-04b8-4640-9462-458b448fbd11.png)

I want to get field label  but field label not found in dictionary. guide me where I am wrong.

{'AP': {'D': {'Off': <PDFObjRef:36>, 'Yes': <PDFObjRef:37>}, 'N': {'Off': <PDFObjRef:34>, 'Yes': <PDFObjRef:35>}}, 'AS': /'Off', 'DA': b'/ZaDb 0 Tf 0 g', 'F': 4, 'FT': /'Btn', 'MK': {'BC': [0.0], 'BG': [1.0], 'CA': b'4'}, 'P': <PDFObjRef:15>, 'Rect': [48.1091, 493.999, 57.4879, 503.378], 'Subtype': /'Widget', 'T': b'Check Box4', 'Type': /'Annot'}
None
{'AP': {'D': {'Off': <PDFObjRef:24>, 'Yes': <PDFObjRef:25>}, 'N': {'Off': <PDFObjRef:22>, 'Yes': <PDFObjRef:23>}}, 'AS': /'Off', 'DA': b'/ZaDb 0 Tf 0 g', 'F': 4, 'FT': /'Btn', 'MK': {'BC': [0.0], 'BG': [1.0]}, 'P': <PDFObjRef:15>, 'Rect': [48.1121, 534.368, 57.4909, 543.747], 'Subtype': /'Widget', 'T': b'Check Box1', 'Type': /'Annot'}
None
{'AP': {'D': {'Off': <PDFObjRef:32>, 'Yes': <PDFObjRef:33>}, 'N': {'Off': <PDFObjRef:30>, 'Yes': <PDFObjRef:31>}}, 'AS': /'Off', 'DA': b'/ZaDb 0 Tf 0 g', 'F': 4, 'FT': /'Btn', 'MK': {'BC': [0.0], 'BG': [1.0], 'CA': b'4'}, 'P': <PDFObjRef:15>, 'Rect': [48.1091, 507.581, 57.4879, 516.96], 'Subtype': /'Widget', 'T': b'Check Box3', 'Type': /'Annot'}
None
{'AP': {'D': {'Off': <PDFObjRef:28>, 'Yes': <PDFObjRef:29>}, 'N': {'Off': <PDFObjRef:26>, 'Yes': <PDFObjRef:27>}}, 'AS': /'Off', 'DA': b'/ZaDb 0 Tf 0 g', 'F': 4, 'FT': /'Btn', 'MK': {'BC': [0.0], 'BG': [1.0], 'CA': b'4'}, 'P': <PDFObjRef:15>, 'Rect': [48.1091, 520.947, 57.4879, 530.326], 'Subtype': /'Widget', 'T': b'Check Box2', 'Type': /'Annot'}
None
{'DA': b'/Arial 12 Tf 0 g', 'F': 4, 'FT': /'Tx', 'Ff': 12582912, 'MK': {}, 'P': <PDFObjRef:15>, 'Rect': [97.6908, 490.399, 552.604, 506.435], 'Subtype': /'Widget', 'T': b'Text5', 'TU': b'Please enter your favourite activity here', 'Type': /'Annot'}
None
{'DA': b'/Arial 12 Tf 0 g', 'FT': /'Btn', 'Ff': 65536, 'Kids': [<PDFObjRef:85>, <PDFObjRef:86>], 'T': b'Button7', 'TU': b'Click here to save this document'}
None
{'DA': b'/Arial 12 Tf 0 g', 'F': 4, 'FT': /'Tx', 'Ff': 12582912, 'MK': {}, 'P': <PDFObjRef:15>, 'Rect': [96.8703, 322.017, 551.784, 338.054], 'Subtype': /'Widget', 'T': b'Text6', 'TU': b'Please enter your favourite activity here', 'Type': /'Annot'}
None
{'DA': b'/ZaDb 0 Tf 0 g', 'FT': /'Btn', 'Ff': 49152, 'Kids': [<PDFObjRef:80>, <PDFObjRef:81>, <PDFObjRef:82>, <PDFObjRef:83>], 'Opt': [b'Choice1', b'Choice1', b'Choice1', b'Choice1'], 'T': b'Group6'}
None

My code is

```python
import pdfminer
from pdfminer.pdfparser import PDFParser
from pdfminer.pdfdocument import PDFDocument
from pdfminer.pdftypes import resolve1
from pdfminer.psparser import PSLiteral, PSKeyword
from pdfminer.utils import decode_text
with open('sample_pdf.pdf', 'rb') as fp:
    parser = PDFParser(fp)
    doc = PDFDocument(parser)
    res = resolve1(doc.catalog)
    if 'AcroForm' not in res:
        raise ValueError("No AcroForm Found")
    fields = resolve1(doc.catalog['AcroForm'])['Fields'] 
    for f in fields:
        field = resolve1(f)
        print(field)
        print(field.get('kid'))
        name, values = field.get('T'), field.get('V')
```

**Comment by MartinThoma** [comment]
Can you share the PDF?

**Reply by Waqarcode** [reply]
[sample_pdf.pdf](https://github.com/py-pdf/PyPDF2/files/8977147/sample_pdf.pdf)
